nautobot/dcim/filters.py

```diff
@@ -327,17 +337,55 @@ class Meta:
         model = Location
         fields = ["id", "name", "slug", "description"]
 
+    def generate_query__base_site(self, value):
+        """Helper method used by DynamicGroups and by _base_site() method."""
+        if value:
+            max_depth = max(loc.tree_depth for loc in Location.objects.with_tree_fields())
```

Comment:
```suggestion
            max_depth = Location.objects.with_tree_fields().extra(order_by=["-__tree.tree_depth"]).first().tree_depth
```
To change this to a `LIMIT 1` query